docs/predictive-analytics/next-best-offer/index.md

```diff
@@ -0,0 +1,214 @@
+---
+seo:
+  title: Next Best Offer — Anticipating Your Customer's Needs 
+  description: This recipe demonstrates how to use ksqlDB to present relevant offers to banking customers.
+---
+
+# Next Best Offer — Anticipating Your Customer's Needs
+
+Consumers today are faced with never-ending marketing messages from a variety of sources. Often these messages are generic and don't have any consideration for the individual needs of the consumer. This one-size-fits-all approach leads to poor conversion rates. A better approach is to tailor offerings that take into consideration the interests of the consumer, based on previous purchases or behavior. This recipe demonstrates how to take existing customer information and provide a "Next Best Offer" to encourage sales and retain customers.
+
+## Step-by-step
+
+### Set up your environment
+
+Provision a Kafka cluster in [Confluent Cloud](https://www.confluent.io/confluent-cloud/tryfree/?utm_source=github&utm_medium=ksqldb_recipes&utm_campaign=next_best_effort).
+
+
+--8<-- "docs/shared/ccloud_setup.md"
+
+### Read the data in
+
+--8<-- "docs/shared/connect.md"
+
+```json
+--8<-- "docs/predictive-analytics/next-best-offer/source.json"
+```
+
+--8<-- "docs/shared/manual_insert.md"
+
+### Run the stream processing application
+
+This application will perform a series of joins between event streams and tables to calculate the next best offer for a banking consumer, based on their activity. This should yield higher customer activity and satisfaction.
+
+--8<-- "docs/shared/ksqlb_processing_intro.md"
+
+```sql
+--8<-- "docs/predictive-analytics/next-best-offer/process.sql"
+```
+
+--8<-- "docs/shared/manual_cue.md"
+
+```sql
+--8<-- "docs/predictive-analytics/next-best-offer/manual.sql"
+```
+
+
+### Cleanup
+
+--8<-- "docs/shared/cleanup.md"
+
+## Explanation
+
+### Creating an event stream
+
+To get started, you'll first need to create a stream that contains the customer activity:
+
+```sql
+CREATE STREAM customer_activity_stream (
+    ACTIVITY_ID INTEGER,
+    IP_ADDRESS STRING,
+    CUSTOMER_ID INTEGER KEY,
+    ACTIVITY_TYPE STRING,
+    PROPENSITY_TO_BUY DOUBLE
+   ) WITH (
+    KAFKA_TOPIC = 'CUSTOMER_ACTIVITY_STREAM',
+    VALUE_FORMAT = 'JSON',
+    PARTITIONS = 6
+);
+```
+
+Note that the `CUSTOMER_ID` field is the key in the stream's key-value pairs. You'll see why this is important in the next section.
+
+In a production setting, you would populate the stream's underlying topic either using a `KafkaProducer` application or from an external system via a [managed connector on Confluent Cloud](https://docs.confluent.io/cloud/current/connectors/index.html). For the purposes of this example, you'll manually insert records into the stream, using [INSERT VALUES](https://docs.ksqldb.io/en/latest/developer-guide/ksqldb-reference/insert-values/#insert-values) statements:
```

Comment:
Even "for the purposes of this example", the primary path should be to use the connector.  Can you change the wording to reflect that `INSERT INTO` is only if no connector can be run in the user's case?